op-scim-bridge/templates/ingress.yaml

```diff
@@ -29,25 +29,23 @@ spec:
   {{- if and .Values.scim.ingress.className (semverCompare ">=1.18-0" .Capabilities.KubeVersion.GitVersion) }}
   ingressClassName: {{ .Values.scim.ingress.className }}
   {{- end }}
-  {{- if .Values.scim.ingress.tls }}
+  {{- with .Values.scim.ingress }}
+  {{- if .tls.enabled }}
   tls:
-    {{- range .Values.scim.ingress.tls }}
-    - hosts:
-        {{- range .hosts }}
-        - {{ . | quote }}
-        {{- end }}
-      secretName: {{ .secretName }}
-    {{- end }}
+    - hosts: 
+        - {{ .domain | quote }}
+      # TODO: Figure out why templating this in the values file doesn't work
+      secretName: "{{ $.Chart.Name }}-bridge-tls"
+  {{- end }}
   {{- end }}
   rules:
-    {{- range .Values.scim.ingress.hosts }}
-    - host: {{ .host | quote }}
+    {{- if .Values.scim.ingress.domain }}
+    - host: {{ .Values.scim.ingress.domain | quote }}
       http:
         paths:
-          {{- range .paths }}
-          - path: {{ .path }}
-            {{- if and .pathType (semverCompare ">=1.18-0" $.Capabilities.KubeVersion.GitVersion) }}
-            pathType: {{ .pathType }}
```

Comment:
These are thought provoking changes.

Can you speak a little more to your intention around these changes? Why strip out the range of hosts?

The range was introduced by another PR which at the time convinced us of the utility.

Reply:
The spec actually only accepts a single entry https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#ingressrule-v1-networking-k8s-io

I think what the previous one was going for was a difference in path based vs. host based routing. You can do either, but not both. Keeping the range in makes enumerating the sub-keys a lot harder though. Just my 2 cents.

Reply:
I should clarify:
I know a lot of helm charts get a bit "lazy" and just stub out the entire ingress template, expecting the user to provide their own. Personally I find that a little annoying, but for new users especially it is helpful to simply provide keys (with defaults) for the major required ingress fields and construct it ourselves.